gensim/similarities/docsim.py

```diff
@@ -563,6 +564,107 @@ def __str__(self):
         return "%s<%i docs, %i features>" % (self.__class__.__name__, len(self), self.index.shape[1])
 
 
+class SoftCosineSimilarity(interfaces.SimilarityABC):
+    """Document similarity (like MatrixSimilarity) that uses Soft Cosine Similarity as a similarity
+    measure.
+
+    Parameters
+    ----------
+    corpus: list of lists of (int, float) two-tuples
+        A list of documents in the gensim document format.
+    similarity_matrix : scipy.sparse.csc_matrix
+        A term similarity matrix.
+    num_best : int
+        The number of results to retrieve for a query.
+
+    See Also
+    --------
+    gensim.models.keyedvectors.EuclideanKeyedVectors.similarity_matrix
+        A term similarity matrix produced from term embeddings.
+    gensim.matutils.softcossim
+        The Soft Cosine Similarity.
+
+    Examples
+    --------
+    >>> from gensim.models import Word2Vec
+    >>> from gensim.similarities import SoftCosineSimilarity
+    >>> from gensim.utils import simple_preprocess
+    >>> from gensim.corpora import Dictionary
+    >>> # Given a document collection "corpus", train a word2vec model.
+    >>> model = Word2Vec(corpus, workers=3, size=100)
```

Comment:
all variables should be defined (this is really important for examples). You can give small corpus from `gensim.test.utils` or use, for example, this one

```python
import gensim.downloader as api
corpus = api.load("text8")
```

Reply:
Fixed in 3cf154eb8c8cdbb3ff68ce5d67ac6ba014646130.